python/tk_framework_desktopserver/shotgun/api_v2.py

```diff
@@ -720,14 +719,56 @@ def _compute_sys_path(self):
     # Internal methods
 
     @sgtk.LogManager.log_timing
-    def _cache_actions(self, data, config_data):
+    def _cache_actions(self, data, config_data, cached_contents_hash=None, async=False):
```

Comment:
are we overcomplicating the design by hacing `_cache_actions` both somehow being synchronous and async - by calling itself in another thread? Threading code is always brittle, because threading is brittle, so i think by making code running in threads as small and clear as possible, we reduce the risk for future errors. Maybe this method could be split into several different ones?

Reply:
It seems like it can be split relatively easily. Sounds like a good idea to me!